language/e2e-testsuite/src/tests/parallel_execution.rs

```diff
@@ -121,3 +122,33 @@ fn rotate_ed25519_key() {
         &TransactionStatus::Keep(KeptVMStatus::Executed),
     );
 }
+
+#[test]
+fn parallel_execution_config() {
+    let mut executor = FakeExecutor::from_fresh_genesis();
+    let account_size = 1000usize;
+    let initial_balance = 2_000_000u64;
+    let initial_seq_num = 10u64;
+    let accounts = executor.create_accounts(account_size, initial_balance, initial_seq_num);
+
+    // set up the transactions
+    let transfer_amount = 1_000;
+
+    // insert a block prologue transaction
+    let (txns_info, transfer_txns) = create_cyclic_transfers(&executor, &accounts, transfer_amount);
+
+    executor.enable_parallel_execution();
```

Comment:
I think it might be worth testing a malformed config (unable to deserialize) on-chain and make sure it safely fall back

Reply:
Test added!